File: 66-plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        int i = n-1;
        while(i>=0){
            if(digits[i]<9){
                digits[i]+=1;
                return digits;

            }
            if(digits[i]==9){
                digits[i] = 0;
                i--;
            }
  
         }
      int[] newDigits = new int[n + 1];
        newDigits[0] = 1;  
return newDigits;    }
}

File: 66-plus-one/plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let n = digits.length;
    let i = n-1;
    while(i>=0){
     if(digits[i]<9){
                digits[i]+=1;
                return digits;
            }

            if(digits[i]==9){
                digits[i]=0;
                i--;
            }     }
   digits.unshift(1);
   return digits;
    
};